clients/fides-js/src/fides-tcf.ts

```diff
@@ -0,0 +1,131 @@
+ * - initialize the page with default consent options (e.g. opt-out of advertising cookies, opt-in to analytics, etc.)
+ * - read/write the current user's consent preferences to their browser as a cookie
+ * - push the current user's consent preferences to other systems via integrations (Google Tag Manager, Meta, etc.)
+ *
+ * See https://fid.es for more information!
+ *
+ * Basic usage of this module in an HTML page is:
+ * ```
+ * <script src="https://privacy.{company}.com/fides.js"></script>
+ * <script>
+ *   window.Fides.init({
+ *     consent: {
+ *       options: [{
+ *         cookieKeys: ["data_sales"],
+ *         default: true,
+ *         fidesDataUseKey: "advertising"
+ *       }]
+ *     },
+ *     experience: {},
+ *     geolocation: {},
+ *     options: {
+ *           debug: true,
+ *           isDisabled: false,
+ *           isGeolocationEnabled: false,
+ *           geolocationApiUrl: "",
+ *           overlayParentId: null,
+ *           modalLinkId: null,
+ *           privacyCenterUrl: "http://localhost:3000"
+ *         }
+ *   });
+ * </script>
+ * ```
+ *
+ * ...and later:
+ * ```
+ * <script>
+ *   // Query user consent preferences
+ *   if (window.Fides.consent.data_sales) {
+ *     // ...enable advertising scripts
+ *   }
+ * ```
+ */
+import { gtm } from "./integrations/gtm";
+import { meta } from "./integrations/meta";
+import { shopify } from "./integrations/shopify";
+
+import { FidesConfig } from "./lib/consent-types";
+
+import { tcf } from "./lib/tcf";
+import {
+  getInitialCookie,
+  getInitialFides,
+  initialize,
+} from "./lib/initialize";
+import type { Fides } from "./lib/initialize";
+import { dispatchFidesEvent } from "./lib/events";
+import { isNewFidesCookie } from "./fides";
+
+// The global Fides object; this is bound to window.Fides if available
+// eslint-disable-next-line no-underscore-dangle,@typescript-eslint/naming-convention
+let _Fides: Fides;
+
+/**
+ * Initialize the global Fides object with the given configuration values
+ */
+const init = async (config: FidesConfig) => {
+  const cookie = getInitialCookie(config);
+  const initialFides = getInitialFides({ ...config, cookie });
+  if (initialFides) {
+    Object.assign(_Fides, initialFides);
+    dispatchFidesEvent("FidesInitialized", cookie, config.options.debug);
+    dispatchFidesEvent("FidesUpdated", cookie, config.options.debug);
+  }
+  const updatedFides = await initialize({ ...config, cookie });
+  Object.assign(_Fides, updatedFides);
+
+  // Dispatch the "FidesInitialized" event to update listeners with the initial
+  // state. Skip if we already initialized due to an existing cookie.
+  // For convenience, also dispatch the "FidesUpdated" event; this allows
+  // listeners to ignore the initialization event if they prefer
+  if (isNewFidesCookie(cookie)) {
+    dispatchFidesEvent("FidesInitialized", cookie, config.options.debug);
+  }
+  dispatchFidesEvent("FidesUpdated", cookie, config.options.debug);
+
+  tcf();
```

Comment:
this line is the only difference between `fides.ts` and `fides-tcf.ts` right now, but we could certainly add more.

This refactor was a bit difficult since we use this `_Fides` global variable which stays up to date with `window.Fides`. that's not functional, and so it's hard to just pull all of the original initialize logic into functions directly, especially if we want to dispatch events after changes to the global variable. so I broke the initialization logic into a few functions which return updates to the Fides object, and then this function is in charge of applying those updates (via `Object.assign`)